src/details/ArborX_DetailsBoundingVolumeHierarchyImpl.hpp

```diff
@@ -408,17 +504,23 @@ void BoundingVolumeHierarchyImpl<DeviceType>::queryDispatch(
     // [ A0 A1 B0 B1 C0 C1 ... X0 X1 ]
     //   ^     ^     ^         ^     ^
     //   0     2     4         2N-2  2N
-    reallocWithoutInitializing(indices, n_results);
+    reallocWithoutInitializing(out, n_results);
     Kokkos::parallel_for(
         ARBORX_MARK_REGION("second_pass"),
         Kokkos::RangePolicy<ExecutionSpace>(0, n_queries),
         KOKKOS_LAMBDA(int i) {
           int count = 0;
+          auto const shift = offset(permute(i));
+          auto const &query = queries(i);
           Details::TreeTraversal<DeviceType>::query(
-              bvh, queries(i),
-              [indices, offset, permute, i, &count](int index) {
-                indices(offset(permute(i)) + count++) = index;
+              bvh, query, [&query, &callback, &out, shift, &count](int index) {
+                callback(query, index,
+                         [&count, &out,
+                          shift](typename OutputView::value_type const &value) {
+                           out(shift + count++) = value;
+                         });
               });
+          assert(offset(permute(i) + 1) - offset(permute(i)) == count);
```

Comment:
We have `ARBORX_ASSERT` instead of `assert`

Reply:
Can't do.  This is device code.